pkg/scheduler/framework/zz_generated.deepcopy.go

```diff
@@ -0,0 +1,173 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by deepcopy-gen. DO NOT EDIT.
+
+package framework
+
+import (
+	v1 "k8s.io/api/core/v1"
+	sets "k8s.io/apimachinery/pkg/util/sets"
+)
+
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
+func (in *AffinityTerm) DeepCopyInto(out *AffinityTerm) {
+	*out = *in
+	if in.Namespaces != nil {
+		in, out := &in.Namespaces, &out.Namespaces
+		*out = make(sets.Set[string], len(*in))
+		for key, val := range *in {
+			(*out)[key] = val
+		}
+	}
+	if in.Selector != nil {
+		out.Selector = in.Selector.DeepCopySelector()
+	}
+	if in.NamespaceSelector != nil {
+		out.NamespaceSelector = in.NamespaceSelector.DeepCopySelector()
+	}
+	return
+}
+
+// DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new AffinityTerm.
+func (in *AffinityTerm) DeepCopy() *AffinityTerm {
+	if in == nil {
+		return nil
+	}
+	out := new(AffinityTerm)
+	in.DeepCopyInto(out)
+	return out
+}
+
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
+func (in *ClusterEvent) DeepCopyInto(out *ClusterEvent) {
+	*out = *in
+	return
+}
+
+// DeepCopy is an autogenerated deepcopy function, copying the receiver, creating a new ClusterEvent.
+func (in *ClusterEvent) DeepCopy() *ClusterEvent {
+	if in == nil {
+		return nil
+	}
+	out := new(ClusterEvent)
+	in.DeepCopyInto(out)
+	return out
+}
+
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
+func (in *PodInfo) DeepCopyInto(out *PodInfo) {
+	*out = *in
+	if in.Pod != nil {
+		in, out := &in.Pod, &out.Pod
+		*out = new(v1.Pod)
+		(*in).DeepCopyInto(*out)
+	}
+	if in.RequiredAffinityTerms != nil {
+		in, out := &in.RequiredAffinityTerms, &out.RequiredAffinityTerms
+		*out = make([]AffinityTerm, len(*in))
+		for i := range *in {
+			(*in)[i].DeepCopyInto(&(*out)[i])
+		}
+	}
+	if in.RequiredAntiAffinityTerms != nil {
+		in, out := &in.RequiredAntiAffinityTerms, &out.RequiredAntiAffinityTerms
+		*out = make([]AffinityTerm, len(*in))
+		for i := range *in {
+			(*in)[i].DeepCopyInto(&(*out)[i])
+		}
+	}
+	if in.PreferredAffinityTerms != nil {
+		in, out := &in.PreferredAffinityTerms, &out.PreferredAffinityTerms
+		*out = make([]WeightedAffinityTerm, len(*in))
+		for i := range *in {
+			(*in)[i].DeepCopyInto(&(*out)[i])
+		}
+	}
+	if in.PreferredAntiAffinityTerms != nil {
+		in, out := &in.PreferredAntiAffinityTerms, &out.PreferredAntiAffinityTerms
+		*out = make([]WeightedAffinityTerm, len(*in))
+		for i := range *in {
+			(*in)[i].DeepCopyInto(&(*out)[i])
+		}
+	}
+	if in.cachedResource != nil {
+		in, out := &in.cachedResource, &out.cachedResource
+		*out = new(podResource)
+		(*in).DeepCopyInto(*out)
```

Comment:
> (*in).DeepCopyInto undefined (type *podResource has no field or method DeepCopyInto)

https://github.com/kubernetes/kubernetes/blob/a1a4bc8fec8d5e1e682d85638508dc1381345508/pkg/scheduler/framework/types.go#L426-L430

the podResource struct is a private struct, so we can't mark it with `// +k8s:deepcopy-gen=true`. 

I tried to rename it to `PodResource`, but the `non0CPU` and `non0Mem` fields are missing in the generated code because they are private fields and their types are bulit-in types.

It looks like we have to make the `podResource` struct public and all fields are exported. cc @sanposhiho